src/smt/smt_engine.cpp

```diff
@@ -238,10 +238,13 @@ void SmtEngine::finishInit()
   }
 
   Trace("smt-debug") << "SmtEngine::finishInit" << std::endl;
-  // if proofs and unsat cores, proofs are used solely for unsat core
-  // production, so we don't generate proofs in the theory engine, which is
-  // communicated via the second argument
-  d_smtSolver->finishInit(logic, options::unsatCoresNew());
+  // if proofs and unsat cores, proofs are used solely for unsat core production
+  // unless the unsat cores are in FULL_PROF mode. Thus we don't generate proofs
+  // in the theory engine, which is communicated via the second argument
+  d_smtSolver->finishInit(
+      logic,
+      options::unsatCoresNew()
```

Comment:
I think it may be better to just push this logic into SmtSolver.

The issue here is that this Boolean value isn't being used in a central way, thus SmtSolver::finishInit is probably a more appropriate place to query the options.